<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TextureEditorDialog</class>
 <widget class="QDialog" name="TextureEditorDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>760</width>
    <height>420</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>640</width>
    <height>360</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Texture Editor</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* ============================================================
   TextureEditorDialog stylesheet (MaterialEditor-like)
   ============================================================ */

QDialog {
    background-color: #1b1d20;
}

/* Buttons */
QPushButton {
    background-color: rgb(35, 35, 37);
    border: none;
    color: rgb(127, 137, 148);
    padding: 1px 0px;
    text-align: center;
    font-size: 9pt;
    line-height: 23px;
    margin: 0px;
}
QPushButton:pressed { background-color: rgb(68, 93, 132); }
QPushButton:hover  { background-color: rgb(35, 35, 36); }
QPushButton:checked {
    background-color: rgb(68, 93, 132);
    color: white;
}

/* Inputs */
QLineEdit {
    background-color: #282b2e;
    border: none;
    padding-left: 6px;
    color: rgb(200, 205, 210);
    min-height: 22px;
}
QComboBox {
    background-color: #282b2e;
    border: none;
    border-radius: 0px;
    padding-left: 6px;
    color: rgb(200, 205, 210);
    min-height: 22px;
}
QComboBox::drop-down { border: none; }
QComboBox QAbstractItemView {
    background-color: #282b2e;
    color: white;
    border: none;
}

/* List */
QListWidget {
    border: 0px;
    outline: none;
    background: transparent;
}
QListWidget::item {
    border-radius: 0px;
    margin-left: 1px;
    margin-right: 1px;
    padding: 0px 8px;
}
QListWidget::item:selected {
    background: rgba(60, 120, 215, 180);
    color: white;
}
QListWidget::item:selected:!active {
    background: rgba(60, 120, 215, 160);
    color: white;
}
QListWidget::item:hover {
    background: rgba(60, 120, 215, 90);
}

/* Splitter (hide handle like MaterialEditor) */
QSplitter {
    background: transparent;
    border: 0px;
    margin: 0px;
    padding: 0px;
}
QSplitter::handle { background: transparent; border: 0px; margin: 0px; padding: 0px; }
QSplitter::handle:horizontal { width: 0px; }
QSplitter::handle:vertical   { height: 0px; }

/* Header/label cells (bluish labels) */
QFrame#topHeaderFrame QLabel,
QFrame#leftHeaderFrame QLabel,
QFrame#rightHeaderFrame QLabel,
QFrame#propsFrame QLabel,
QFrame#footerFrame QLabel {
    background-color: #303645;
    color: rgb(92,102,113);
    font-weight: 550;
    padding: 5px 8px;
    border-radius: 0px;
}

/* Property grid background */
QFrame#propsFrame {
    background-color: #22262c;
}

/* Preview box */
#previewWidgetPlaceholder {
    background-color: #1f232a;
    border: 1px solid #2a2f3a;
    color: rgb(160, 170, 180);
}
</string>
  </property>

  <layout class="QVBoxLayout" name="rootLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>

   <!-- Top header (FIXED HEIGHT so it won't stretch on resize) -->
   <item>
    <widget class="QFrame" name="topHeaderFrame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>25</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>25</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::Shape::StyledPanel</enum>
     </property>
     <layout class="QHBoxLayout" name="topHeaderLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="titleLabel">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>25</height>
         </size>
        </property>
        <property name="text">
         <string>Texture Editor</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeft|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>

   <!-- Main splitter (this is the ONLY thing that should grow vertically) -->
   <item>
    <widget class="QSplitter" name="splitterMain">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <property name="childrenCollapsible">
      <bool>false</bool>
     </property>

     <!-- Left panel -->
     <widget class="QWidget" name="leftPanel">
      <layout class="QVBoxLayout" name="leftLayout">
       <property name="spacing">
        <number>1</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>

       <item>
        <widget class="QFrame" name="leftHeaderFrame">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>25</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>25</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::StyledPanel</enum>
         </property>
         <layout class="QHBoxLayout" name="leftHeaderLayout">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="textureListLabel">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>25</height>
             </size>
            </property>
            <property name="text">
             <string>Texture List</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>

       <item>
        <widget class="QListWidget" name="textureListWidget">
         <property name="minimumSize">
          <size>
           <width>220</width>
           <height>0</height>
          </size>
         </property>
         <property name="uniformItemSizes">
          <bool>true</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>

     <!-- Right panel -->
     <widget class="QWidget" name="rightPanel">
      <layout class="QVBoxLayout" name="rightLayout">
       <property name="spacing">
        <number>1</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>

       <item>
        <widget class="QFrame" name="rightHeaderFrame">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>25</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>25</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::StyledPanel</enum>
         </property>
         <layout class="QHBoxLayout" name="rightHeaderLayout">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="rightTitleLabel">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>25</height>
             </size>
            </property>
            <property name="text">
             <string>Texture Properties</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>

       <!-- Preview -->
       <item>
        <layout class="QVBoxLayout" name="previewLayout">
         <property name="spacing">
          <number>1</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="previewLabel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>18</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>18</height>
            </size>
           </property>
           <property name="text">
            <string>Preview</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="previewWidgetPlaceholder">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>220</height>
            </size>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </item>

       <!-- Properties grid (FIXED HEIGHT so it won't stretch) -->
       <item>
        <widget class="QFrame" name="propsFrame">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <!-- 6 rows * 25 + 5 spacers * 1 = 155 -->
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>155</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>155</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::NoFrame</enum>
         </property>
         <layout class="QGridLayout" name="propsGrid">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <property name="spacing">
           <number>1</number>
          </property>

          <item row="0" column="0">
           <widget class="QLabel" name="nameLabel">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>25</height>
             </size>
            </property>
            <property name="text">
             <string>Name</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="nameLineEdit"/>
          </item>

          <item row="1" column="0">
           <widget class="QLabel" name="pathLabel">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>25</height>
             </size>
            </property>
            <property name="text">
             <string>Path</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="pathLineEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>

          <item row="2" column="0">
           <widget class="QLabel" name="sizeLabel">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>25</height>
             </size>
            </property>
            <property name="text">
             <string>Size</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="sizeValueLabel">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="text">
             <string>-</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>

          <item row="3" column="0">
           <widget class="QLabel" name="formatLabel">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>25</height>
             </size>
            </property>
            <property name="text">
             <string>Format</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="formatValueLabel">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="text">
             <string>-</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>

          <item row="4" column="0">
           <widget class="QLabel" name="colorSpaceLabel">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>25</height>
             </size>
            </property>
            <property name="text">
             <string>Color Space</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QComboBox" name="colorSpaceCombo">
            <item>
             <property name="text">
              <string>sRGB</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Linear</string>
             </property>
            </item>
           </widget>
          </item>

          <item row="5" column="0">
           <widget class="QLabel" name="usedByLabel">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>25</height>
             </size>
            </property>
            <property name="text">
             <string>Used By</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QLabel" name="usedByValueLabel">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="text">
             <string>-</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>

         </layout>
        </widget>
       </item>

       <!-- This spacer eats ALL extra height so things don't "spread" -->
       <item>
        <spacer name="rightBottomSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Policy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>

      </layout>
     </widget>
    </widget>
   </item>

   <!-- Footer (FIXED HEIGHT so it won't stretch) -->
   <item>
    <widget class="QFrame" name="footerFrame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>25</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>25</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::Shape::NoFrame</enum>
     </property>
     <layout class="QHBoxLayout" name="footerLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>

      <item>
       <widget class="QPushButton" name="addButton">
        <property name="minimumSize">
         <size>
          <width>26</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>26</width>
          <height>25</height>
         </size>
        </property>
        <property name="text">
         <string>+</string>
        </property>
       </widget>
      </item>

      <item>
       <widget class="QPushButton" name="removeButton">
        <property name="minimumSize">
         <size>
          <width>26</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>26</width>
          <height>25</height>
         </size>
        </property>
        <property name="text">
         <string>-</string>
        </property>
       </widget>
      </item>

      <item>
       <spacer name="footerSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Policy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>

      <item>
       <widget class="QPushButton" name="closeButton">
        <property name="minimumSize">
         <size>
          <width>90</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>90</width>
          <height>25</height>
         </size>
        </property>
        <property name="text">
         <string>Close</string>
        </property>
       </widget>
      </item>

     </layout>
    </widget>
   </item>

  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
